### ISSUE TITLE
visibilty of q-layout-drawer doesnt work after reload if :overlay=true is set.

### ISSUE THREAD

**Issue by ibrainventures** [quasarframework/quasar#1970]
if i use the q-layout-drawer with the visibility control by the v-model (false / true) it
works fine .. also with reload.

If i add the :overlay="true" option the visibilty control by the v-model is not working
after the reload ... (set to true, reload ... no drawer occurs.)

this problem is also at some other users (maiko, ..) we discorded this :-)

Quasar: 0.15.10
-> all browsers  
  
<q-layout-drawer
      v-model="leftDrawerOpen"
      :content-class="$q.theme === 'mat' ? 'bg-grey-2' : null"
      :overlay="true"
      side="left"
    >

**Comment by pdanpdan** [comment]
Please create a fiddle so we can test the problem.

**Comment by ibrainventures** [comment]
hi pdanpdan,

i tried to simulate this on a jsfiddle, but the problem is, to show this reload issue - the fiddle framework is capturing the reload - so i cannot give you a fiddle.

for you, to simulate this issue : add the :overlay="true" and make a v-model true this will be a 20 second simulation ...

if this works on your test enviroement - myself and all other issued-users can forget this issue ..

sorry for this ..

**Comment by pdanpdan** [comment]
Ah, I got it. This is so by design. If you need you can set model to true with a small delay in `mounted` ($nextTick should be enough I think).

**Comment by pdanpdan** [comment]
Please make a jsfiddle so I can check your code.

**Comment by emahuni** [comment]
@pdanpdan I think this is unexpected behaviour and undesired, I understand why this is the case now that you mention it, but it's too opinionated and not intuitive as to what the documentation infers. For instance, I have a menu that needs to be visible at all times. It is set to be visible with `:value='true'`, which is intuitive since I don't want it controlled by any prop, then it should show as expected. And having the drawer then hide by default because of overlay and without knowledge of the developer is actually a nightmare to say the least. At least this should appear in the documentation. But the correct thing would just do what it should do, just as what developers expect, show up when asked to show up regardless of when, as the documentation says. I thought v-model and value primarily control it's visibility in addition to breakpoint, behavior, and persistent props, not overlay, this property should just tell the drawer to be above things and not meddle with visibility. Otherwise someone spends hours hunting a bug that's not even in their code and then somehow ends up in Quasar Github to only find out that it is by an undocumented design principle. 😅

**Comment by emahuni** [comment]
It then becomes a really headache bug when you reload. Whoaaaa! where did it go?

**Comment by justice47** [comment]
I'm agree with @emahuni , but I'm faced with that problem even without overlay - 
```
 <q-drawer v-model="left" side="left" behavior="desktop" persistent elevated>
```
even if `left` set to `true` drawer is hided at the page load. Why does is so?
Should I open new issue?

**Comment by hawkeye64** [comment]
Please upgrade to latest Quasar (v1.0.5 atm)

**Comment by rstoenescu** [comment]
I keep explaining this over and over again why the default state needs to go to hidden...
Reopening to give it another go. There's lots of scenarios that need to be taken into account and very smart solutions needed for some. Will keep you all updated on the progress.

**Comment by rstoenescu** [comment]
QDrawer will no longer tamper with the v-model, but this means to take into account to initialize the v-model correctly (usually `model: this.$q.screen.width >= 992`, where you replace 992 by your drawer breakpoint --- 992 is the default one)

This change will be available in 1.1.0.

**Comment by ibrainventures** [comment]
as issuer opener i give a big "Thank you" for optimizing this behaviour  ..
